### PR TITLE
feat: demo navigate out

### PR DESCRIPTION
**Pull request by simboonlong**
- https://www.pivotaltracker.com/story/show/179399657
- demo flow to show prompt modal, when user navigate out

### CODE REVIEW

**Comment by cavacado** [comment]
You think we want to handle the navigate out to external site?
I mean if the user wants to navigate out from the demo to some other website so be it then?

**Comment by simboonlong** [comment]
um no idea, im fine with that as well

**Comment by cavacado** [comment]
yup, lets just not handle it. Unless Jerry thinks otherwise.

File: src/components/Demo/DemoCreate/index.tsx [hunk 1]
```diff
@@ -1,15 +1,24 @@
...
 import { DemoCreateReview } from "./DemoCreateReview";
 import { DemoCreateStart } from "./DemoCreateStart";
+import { ModalNavigateOut } from "../ModalNavigateOut";
 
 export const DemoCreate: FunctionComponent = () => {
+  const history = useHistory();
   const { activeStep } = useContext(DemoCreateContext);
+  const { setOverlayVisible, showOverlay } = useOverlayContext();
+  const [isUserNavigateOut, setIsUserNavigateOut] = useState(false);
+  const when = activeStep !== "start";
```
Comment (cavacado) [inline src/components/Demo/DemoCreate/index.tsx:21]: ~can we rename `when` to something more descriptive? maybe `onNavigateOut`?~
zz, when i think about it its just makes things more confusing

**Review by cavacado — APPROVED** [review]
still thinking whehther we should put the `Prompt` in MagicDemoDropZone since its actually part of the entire flow.
but will approve first.

**Pull request merged**